src/short-urls/short-urls.service.ts

```diff
@@ -0,0 +1,50 @@
+import { Injectable } from '@nestjs/common';
+import { CreateShortUrlDto } from '@/short-urls/dto/create-short-url.dto';
+import { errorMessage } from '@/common/messages';
+import { ShortUrl } from '@/short-urls/entities/short-url.entity';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository, TypeORMError } from 'typeorm';
+import { generateUrlCode } from '@/short-urls/util/generate-url-code';
+import { User } from '@/users/entities/user.entity';
+import { LoggerService } from '@/logger/logger.service';
+@Injectable()
+export class ShortUrlsService {
+	constructor(
+		private readonly logger: LoggerService,
+		@InjectRepository(ShortUrl)
+		private shortUrlRepository: Repository<ShortUrl>,
+	) {}
+	async createShortUrl(user: User, { originalUrl, expiryDate }: CreateShortUrlDto): Promise<Partial<ShortUrl>> {
+		const urlCode = await this.generateUniqueCode();
+		const shortUrl = {
+			user_id: user.id,
+			original_url: originalUrl,
+			expires_at: expiryDate,
+			short_code: urlCode,
+		};
+		const result = await this.shortUrlRepository.insert(shortUrl);
+		if (!result) {
+			throw new TypeORMError(errorMessage.urlCreationFailure);
+		}
+		this.logger.log(`${user.email} created a new short URL`);
```

Comment:
You can instead do 
```suggestion
               const shortUrlEntity = this.shortUrlRepository.create(shortUrl);
               const savedUrl = await this.shortUrlRepository.save(shortUrlEntity);
		if (!result) {
			throw new TypeORMError(errorMessage.urlCreationFailure);
		}
		this.logger.log(`${user.email} created a new short URL with id ${result.id}`);
```

The reason is that `.insert` returns an InsertResponse, which does not contain the autogenerated ID. When debugging with logs, you need to add a unique identifier wherever possible so all affected entities can be tracked

Reply:
This was already discussed previously, when using save, typeorm first checks existing record in the database and then only inserts/update which hampers performance.

Reply:
@rrojan  I think i didn't understand what you mean by containing autogenerated ID , but insert does contain ID within `generated` array but I haven't checked whether it is of the respective user that was created or not.